### PR TITLE
ENH: GH11128 add weekday_name to DatetimeIndex and .dt

### LINKED ISSUE #11128: ENH: add weekday_name to DatetimeIndex and .dt
These names exist already in ``pd.tslib`` (as the below weedays calc), should just create them as a variable so can import it directly here.

```
In [1]: import calendar
In [25]: s = pd.Series(list(range(10)), pd.date_range('2015-09-01', '2015-09-10'))

In [26]: s2 = s.to_frame('value').assign(weekday=Series(s.index.weekday).map(Series(weekdays)).values)

In [27]: weekdays = [calendar.day_name[i].lower() for i in range(7)]

In [28]: s2 = s.to_frame('value').assign(weekday=Series(s.index.weekday).map(Series(weekdays)).values)

In [29]: s2
Out[29]: 
            value    weekday
2015-09-01      0    tuesday
2015-09-02      1  wednesday
2015-09-03      2   thursday
2015-09-04      3     friday
2015-09-05      4   saturday
2015-09-06      5     sunday
2015-09-07      6     monday
2015-09-08      7    tuesday
2015-09-09      8  wednesday
2015-09-10      9   thursday
```

### PR DESCRIPTION
**Pull request by BastiaanBergman**
closes #11128 
replaces #11813
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

Completed merge of Bahrunnur's PR 'weekday-name'
    Moved 'enhancements' to v0.18.1.txt
    Added tests to pandas/tests/series/test_datetime_values.py
    Added weekday_name property producing 'NaT' to NaTType class

### CODE REVIEW

File: doc/source/api.rst [hunk 1]
```diff
@@ -469,6 +469,7 @@ These can be accessed like ``Series.dt.<property>``.
    Series.dt.days_in_month
    Series.dt.tz
    Series.dt.freq
+   Series.dt.weekday_name
```
Comment (jreback) [inline doc/source/api.rst]: can you move after ``.weekday``

File: doc/source/api.rst [hunk 2]
```diff
@@ -1487,6 +1488,7 @@ Time/Date Components
    DatetimeIndex.is_year_start
    DatetimeIndex.is_year_end
    DatetimeIndex.inferred_freq
+   DatetimeIndex.weekday_name
```
Comment (jreback) [inline doc/source/api.rst]: same here

File: pandas/tseries/tests/test_timeseries.py [hunk 3]
```diff
@@ -3541,6 +3541,15 @@ def test_datetimeindex_accessors(self):
         self.assertEqual(dti.is_year_end[0], False)
         self.assertEqual(dti.is_year_end[364], True)
 
+        # GH 11128
+        self.assertEqual(dti.weekday_name[4], u'Monday')
+        self.assertEqual(dti.weekday_name[5], u'Tuesday')
```
Comment (jreback) [inline pandas/tseries/tests/test_timeseries.py]: can you add a test for ``Timestamp`` (see where tests for ``.weekday`` are, prob in ``tseries/test_timeseries.py``

File: pandas/tslib.pyx [hunk 4]
```diff
@@ -4390,6 +4395,38 @@ def get_start_end_field(ndarray[int64_t] dtindex, object field, object freqstr=N
...
+    cdef:
+        _TSObject ts
+        Py_ssize_t i, count = 0
+        ndarray[object] out
+        pandas_datetimestruct dts
+        int dow
+
+    _dayname = np.array(
+        ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'],
+         dtype=np.str )
```
Comment (jreback) [inline pandas/tslib.pyx]: ``dtype=np.object_`` is more consistent with what we use.

File: pandas/tseries/tests/test_base.py [hunk 5]
```diff
@@ -32,7 +32,7 @@ def test_ops_properties(self):
                                    'is_month_start', 'is_month_end',
                                    'is_quarter_start',
                                    'is_quarter_end', 'is_year_start',
-                                   'is_year_end'],
```
Comment (jreback) [inline pandas/tseries/tests/test_base.py]: there are addl tests in ``tests/series/test_datetime_values.py`` (just add the new propery name)

**Comment by jreback** [comment]
looks good. just some minor points.

come to think of it, prob should add ``month_name`` as well I guess. Then of course there is the locale issue. Can you make an issue for these 2 things?

**Comment by jreback** [comment]
I *think* we can just instantiate the ``LocaleTime()`` object when this module is imported. Not really sure why that wouldn't work, you can try it if you want. This should really be a singleton type object, e.g.

```
locale_time = LocalTime()
```

then use ``locale_time``

File: pandas/tseries/tests/test_base.py (hunk 5 above)
Comment (BastiaanBergman) [inline pandas/tseries/tests/test_base.py]: Yes, and they were added to in my original PR.

File: pandas/tests/series/test_datetime_values.py [hunk 6]
```diff
@@ -25,7 +25,7 @@ class TestSeriesDatetimeValues(TestData, tm.TestCase):
 
     def test_dt_namespace_accessor(self):
 
-        # GH 7207
+        # GH 7207, 11128
```
Comment (jreback) [inline pandas/tests/series/test_datetime_values.py]: I mean add it here as well

File: pandas/tslib.pyx [hunk 7]
```diff
@@ -4411,7 +4411,7 @@ def get_date_name_field(ndarray[int64_t] dtindex, object field):
 
     _dayname = np.array(
         ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'],
-         dtype=np.str )
+         dtype=np.object_ ) # was np.str
```
Comment (jreback) [inline pandas/tslib.pyx]: remove the comment

**Comment by jreback** [comment]
ok this looks good. I'd like to use the ``LocaleTime()`` object as I described (should be cached) to return the locale specific names here.

**Comment by jreback** [comment]
xref #12806

**Comment by jreback** [comment]
can you try adding this with the ``LocaleTime``support?

**Comment by BastiaanBergman** [comment]
Yes, just need to find the time, maybe later this week.

**Comment by jreback** [comment]
@BastiaanBergman ok, great thanks

**Comment by jreback** [comment]
@jorisvandenbossche comments?

I guess we can address using ``LocaleTime()`` later.

**Comment by BastiaanBergman** [comment]
Sorry, didn't get to it, will try again late May.

**Comment by jorisvandenbossche** [comment]
Looks good

**Comment by jreback** [comment]
thanks!

**Pull request closed without merging**